### ISSUE TITLE
Uninitialized var in function Modelica.Media.Water.IF97_Utilities.waterBaseProp_dT used

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1422]
**Reported by jfrenkel on 14 Feb 2014 12:40 UTC**
In the Modelica MSL function Modelica.Media.Water.IF97_Utilities.waterBaseProp_dT is the uninitialized variable aux.cv used before assigned.

source code from trunc(without unnecessary statements)

```
function waterBaseProp_dT "Intermediate property record for water (d and T preferred states)"
	output Common.IF97BaseTwoPhase aux "Auxiliary record" annotation(Placement(transformation(extent={{-10,-10},{10,10}})));
	algorithm
		aux.region := if region == 0 then (if phase == 2 then 4 else
		  BaseIF97.Regions.region_dT(
		    d=rho,
		    T=T,
		    phase=phase)) else region;
		aux.phase := if aux.region == 4 then 2 else 1;
		aux.R := BaseIF97.data.RH2O;
		aux.rho := rho;
		aux.T := T;
		if (aux.region == 1) then
                ...
		elseif (aux.region == 3) then
		  f := BaseIF97.Basic.f3(rho, T);
		  aux.p := aux.R*rho*T*f.delta*f.fdelta;
		  aux.h := aux.R*T*(f.tau*f.ftau + f.delta*f.fdelta);
		  aux.s := aux.R*(f.tau*f.ftau - f.f);
		  aux.pd := aux.R*T*f.delta*(2.0*f.fdelta + f.delta*f.fdeltadelta);
		  aux.pt := aux.R*rho*f.delta*(f.fdelta - f.tau*f.fdeltatau);
		  aux.cp := (aux.rho*aux.rho*aux.pd*aux.cv + aux.T*aux.pt*aux.pt)/(aux.rho*
		    aux.rho*aux.pd);
		  aux.cv := aux.R*(-f.tau*f.tau*f.ftautau);
end waterBaseProp_dT;
```
see the last two statements in source code above:

aux.cp := (aux.rho\*aux.rho\*aux.pd\***aux.cv** + aux.T\*aux.pt\*aux.pt)/(aux.rho\*aux.rho\*aux.pd);
**aux.cv** := aux.R\*(-f.tau\*f.tau\*f.ftautau);

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1422

**Comment by modelica-trac-importer** [comment]
**Modified by jfrenkel on 14 Feb 2014 12:40 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 14 Feb 2014 22:21 UTC**
Fixed in trunk in 3715cf627e896df41faa96358b246e23179937e9. Keeping this open until I have time to merge it to 3.2.x branch

**Comment by modelica-trac-importer** [comment]
**Comment by jfrenkel on 6 Mar 2014 16:52 UTC**
By the way. In that function are some elements of aux are not calculated in all cases. 
For example:

aux.vp: not if aux.region == 3 or 4
aux.x: not if aux.region == 5
aux.pd and aux.pt: not if aux.region == 1 or 2 or 5
aux.dpT: not if aux.region == 1 or 2 or 3 or 5

Regarding to Specification I could only find in Chapter 12 how to handle local variables but not outputs in such cases.
```
The local variables of a function are not automatically initialized to the implicit default values of the data type (Section 12.4.4)[(e.g. 0.0 for Real) for performance reasons. It is the responsibility of the user to provide explicit defaults or to define the values of such variables before they are referenced.]
```

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 21 May 2014 19:01 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 21 May 2014 19:43 UTC**
Resolved in MSL 3.2.1 maintenance branch by 972cc1d91e3a5041077fb1baaecd0055c9c6166a.

**Comment by modelica-trac-importer** [comment]
**Changelog modified by beutlich on 21 May 2014 19:43 UTC**
Fixed unitialized variable aux_cv in function Modelica.Media.Water.IF97_Utilities.waterBaseProp_dT
